src/components/StrikeImage/hooks/useStrikeImage.ts

```diff
@@ -0,0 +1,58 @@
+import { useState } from 'react';
+import {
+  interpolate,
+  interpolateColor,
+  useAnimatedStyle,
+  useSharedValue,
+  withTiming,
+} from 'react-native-reanimated';
+import { Colors } from '../../../theme';
+import type { UseStrikeImageReturnType } from '../types';
+
+const useStrikeImage = (
+  size: number,
+  color: string,
+  onChangeStrike: (strikeVisibility: boolean) => void
+): UseStrikeImageReturnType => {
+  const [isStrikeVisible, setIsStrikeVisible] = useState(false);
+
+  const offSet = useSharedValue(0);
+  const lineOffSet = useSharedValue(0);
+
+  const onPressImage = () => {
+    if (offSet.value === 0) {
+      offSet.value = withTiming(0.3, { duration: 500 });
```

Comment:
For duration you can create one const value and use every place where same value. In feature need to change value then it would be better change at one place only.

const animationDuration = 500;